File: leetcode/0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int result = -1;
        int totalSum = reduce(nums.begin(), nums.end(), 0);
        int sumOfNum = 0;
        for(int i = 0; i < nums.size(); i++) {
            if(i != 0) {
                sumOfNum += nums[i - 1];
            }

            totalSum -= nums[i];

            if(sumOfNum == totalSum) {
                result = i;
                break;
            }
        }

        return result;
    }
};

File: leetcode/0724-find-pivot-index/0724-find-pivot-index.swift
class Solution {
    func pivotIndex(_ nums: [Int]) -> Int {
        if(nums.count == 1) {
            return 0
        }

        let sumOfNums = nums.reduce(0, { (a, b) in 
            a + b
        })

        var rightSum = sumOfNums
        var leftSum = 0
        var result = -1

        for (i, e) in nums.enumerated() {
            if(i != 0) {
                leftSum = leftSum + nums[i - 1]
            }
            
            rightSum = rightSum - e

            if(leftSum == rightSum) {
                result = i
                break;
            }
        }

        return result
    }
}